.travis.yml

```diff
@@ -80,6 +80,22 @@ matrix:
         - go get github.com/bazelbuild/buildtools/buildifier
         - ./ci/travis/bazel-format.sh
 
+    - os: linux
+      env: SANITIZER=1 CC=clang PYTHON=3.5 PYTHONWARNINGS=ignore
+
+      install:
+        - eval `python $TRAVIS_BUILD_DIR/ci/travis/determine_tests_to_run.py`
+        - if [ $RAY_CI_PYTHON_AFFECTED != "1" ]; then exit; fi
+
+        - ./ci/suppress_output ./ci/travis/install-bazel.sh
+        - ./ci/suppress_output ./ci/travis/install-dependencies.sh
+        - export PATH="$HOME/miniconda/bin:$PATH"
+        - ./ci/suppress_output ./ci/travis/install-ray.sh
+
+      script:
+        # Run core worker tests with thread sanitizer
+        - RAY_BAZEL_CONFIG=--config=tsan TSAN_OPTIONS=report_atomic_races=0 ./ci/suppress_output bash src/ray/test/run_core_worker_tests.sh
```

Comment:
nit/subjective, but when passing variables that are permitted to have spaces in their values, I find it more readable/maintainable to wrap them in quotes:

    RAY_BAZEL_CONFIG="--config=tsan" ...

Reply:
fixed